commons/src/main/java/com/powsybl/commons/compress/SafeZipInputStream.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.commons.compress;
+
+import java.io.IOException;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import com.powsybl.commons.io.ForwardingInputStream;
+
+public class SafeZipInputStream extends ForwardingInputStream<ZipInputStream> {
+
+    private int bytesRead;
+    private int maxBytesToRead;
+
+    public SafeZipInputStream(ZipInputStream in, int entryNumber, int maxBytesToRead) throws IOException {
+        super(in);
+        this.maxBytesToRead = maxBytesToRead;
+        for (int i = 0; i < entryNumber; i++) {
+            ZipEntry zipEntry = in.getNextEntry();
```

Comment:
We might want to reuse this for other usecases where several ZipInputStream entries have to be read. This is not really easy to use that way, I think you'd need to create a `new SafeZipInputStream(zin, 1, max)` for each entry?

Adding a `SafeZipInputStream::getNextEntry` method would do the trick, at the cost of adding a `protected T getDelegate()` in `ForwardingInputStream`.